### ISSUE TITLE
[Feature] Enum.GetValues(typeof(non-Enum)) warning

### ISSUE THREAD

**Issue by jzabroski** [DotNetAnalyzers/ReflectionAnalyzers#212]
```csharp
public class AB { }
public enum ABEnum { A = 0, B = 1}
public class Test
{
  public void TestAB()
  {
    foreach (var item in Enum.GetValues(typeof(AB))) // should warn here - syntactically valid C# but will blow up at run-time.  Caused due to developer typing AB instead of ABEnum
    {
      Console.WriteLine(((Enum)item).GetDisplayName());
    }
  }
}
```

**Comment by JohanLarsson** [comment]
For the bonus it is a separate analyzer/diagnostic id, create a separate issue for it if you have time.

**Comment by jzabroski** [comment]
@JohanLarsson Fair enough.  Should there be separate analyze/diagnostic id for Enum.GetNames and other static methods introduced in .NET 4?

**Comment by JohanLarsson** [comment]
I usually err on making too many diagnostics as it enables configuration via ruleset. For issues my preference is as small and many as possible. Have found it fastest to work with. If you are interested in trying your hand at writing analyzers for some of them just leave a note. Ask questions in chat if you happen to get stuck.

**Comment by jzabroski** [comment]
@JohanLarsson  Thanks, I would like that.
